apps/emqx_durable_storage/src/emqx_ds.erl

```diff
@@ -186,7 +187,13 @@
         %% Whether to wait until the message storage has been acknowledged to return from
         %% `store_batch'.
         %% Default: `true'.
-        sync => boolean()
+        sync => boolean(),
+        %% Whether the whole batch given to `store_batch' should be inserted atomically as
+        %% a unit.  Note: the whole batch must be crafted so that it belongs to a single
+        %% shard (if applicable to the backend), as the batch will be split accordingly
+        %% even if this flag is `true'.
+        %% Default: `false'.
+        atomic => boolean()
```

Comment:
This flag apparently is still being "checked" by [DS buffer](https://github.com/emqx/emqx/blob/e42021d31482fe25bc6b3a043dcc1ae7f52b3476/apps/emqx_durable_storage/src/emqx_ds_buffer.erl#L84), but it doesn't do anything.  :thinking: